src/D05/BoardingPass.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+
+namespace aoc.D05
+{
+	public class BoardingPass
+	{
+		private int _seatID;
+
+		public BoardingPass(string code)
+		{
+			if (code.Length != 10)
+				throw new ArgumentException("code must be 10 digits long!");
+
+			_seatID = GetSeatID(code);
+		}
+
+		public int SeatID { get { return _seatID; } }
+
+		public static int PartI(BoardingPass[] passes)
+		{
+			int max = -1;
+
+			foreach (var b in passes)
+			{
+				if (b.SeatID > max)
+					max = b.SeatID;
+			}
+
+			return max;
+		}
+
+		public static int PartII(BoardingPass[] passes)
+		{
+			Array.Sort(passes, (x1, x2) => x1.SeatID.CompareTo(x2.SeatID));
+
+			int mySeatID = -1;
+
+			for (int i = 0; i < passes.Length; i++)
+			{
+				if (i != passes.Length - 1) //last element should not be checked
+				{
+					if ((passes[i + 1].SeatID - passes[i].SeatID) != 1)
+					{
+						mySeatID = passes[i].SeatID + 1;
+						break;
+					}
+				}
+			}
+
+			return mySeatID;
+		}
+
+	  private int GetSeatID(string s)
+		{
+			var s_row = s.Substring(0, 7);
+			var s_column = s.Substring(7);
+
+			return 8 * GetRow(s_row) + GetColumn(s_column);
+		}
+
+		private int GetColumn(string s)
+		{
+			return Locate(s);
+		}
+
+		private int GetRow(string s)
+		{
+			return Locate(s);
+		}
+
+		private int Locate(string s)
+		{
+			var iterations = s.Length;
+			var range = new int[] { 0, (int)Math.Pow(2, iterations) - 1 };  //starting point
+
+			for (int i = 0; i < iterations; i++)
+			{
+				range = UpdateRange(s[i], range);
+			}
+
+			return range[0];
+		}
+
+		private int[] UpdateRange(char c, int[] range)
```

Comment:
I would change the signature to receive low and high instead of this int[] range parameter. It is not obvious from the method signature that range is expected to contain 2 values.

Reply:
The return type I would change to a tuple or a class. It is not obvious from method signature that int[] will always contain two values.

Reply:
example using tuple syntax:
`private Tuple<int, int> UpdateRange(char c, Tuple<int, int> range)`

Reply:
funny that u say all that went thru my mind when doing this. I chose int[] coz i was lazy otherwise I fully agree with what you say n I enjoyed that I thought of the exact same things and even Tuple came to mind. Anyway, I can change it now that the stars are earned.